test/markdown/comment.js

```diff
@@ -31,29 +31,33 @@ describe('Marpit comment plugin', () => {
   htmls.forEach(html => {
     context(`with html option as ${html}`, () => {
       const markdown = md({ html })
+      const extractComments = $cheerio =>
+        $cheerio('*')
+          .contents()
+          .filter(function filterComment() {
+            return this.nodeType === 8
+          })
 
       it('extracts comment and stores to token meta', () => {
         const parsed = markdown.parse(text)
-        const comments = parsed.reduce((arr, token) => {
-          if (token.meta && token.meta.marpitComment)
-            return [...arr, ...token.meta.marpitComment]
-
-          return arr
-        }, [])
+        const comments = parsed.reduce(
+          (arr, token) =>
+            token.type === 'marpit_comment' ? [...arr, token.content] : arr,
+          []
+        )
 
         assert(comments.includes('comment!'))
         assert(comments.includes('supports\nmultiline'))
-        assert(comments.includes('inline'))
-        assert(comments.includes('comment in header'))
+
+        // TODO: Supports inline comment
+        // assert(comments.includes('inline'))
+        // assert(comments.includes('comment in header'))
       })
 
-      it('strips comment in rendering', () => {
+      // TODO: Supports inline comment
+      it.skip('strips comment in rendering', () => {
```

Comment:
The inline comment will become not to work by parsing comment as HTML block.

In the current use case (directive), this change may almost not affect the user. However, we are planning to use HTML comment as presenter notes in Marp. The inline comment will become to be used when a tool integrated with Marpit implements the presentation mode.

Currently, we mark a test case as pending.